source/4_tools/regina/src/regina.cpp

```diff
@@ -1,54 +1,582 @@
+
+                renderer::commands::UpdateConstantBufferCommandDesc update_constant_buffer_command_desc;
+                update_constant_buffer_command_desc.element_index = ri.constant_buffer_index;
+                update_constant_buffer_command_desc.buffer_data = memory::make_blob((rsl::byte*)&obj_constants, rsl::memory_size(sizeof(ObjectConstants)));
+
+                renderer::update_constant_buffer(rsl::move(update_constant_buffer_command_desc), curr_object_cb);
+
+                // Updating constant buffer of the cube so the frame should remain dirty
+                ri.num_frames_dirty = renderer::num_frames_in_flight();
+            }
+        }
+    }
+    //-------------------------------------------------------------------------
+    void update_pass_constant_buffers()
+    {
+        renderer::ResourceSlot curr_pass_cb = get_active_pass_constant_buffer_for_frame(renderer::active_frame()->slot_id());
+
+        const glm::mat4& view = g_regina_ctx.view;
+        const glm::mat4& proj = g_regina_ctx.proj;
+
+        glm::mat4 view_proj = view * proj;
+
+        glm::mat4 inv_view = glm::inverse(view);
+        glm::mat4 inv_proj = glm::inverse(proj);
+        glm::mat4 inv_view_proj = glm::inverse(view_proj);
+
+        g_regina_ctx.pass_constants.view = view;
+        g_regina_ctx.pass_constants.inv_view = inv_view;
+        g_regina_ctx.pass_constants.proj = proj;
+        g_regina_ctx.pass_constants.inv_proj = inv_proj;
+        g_regina_ctx.pass_constants.view_proj = view_proj;
+        g_regina_ctx.pass_constants.inv_view_proj = inv_view_proj;
+
+        f32 window_width = static_cast<f32>(globals::window_info().width);
+        f32 window_height = static_cast<f32>(globals::window_info().height);
+
+        g_regina_ctx.pass_constants.eye_pos_w = g_regina_ctx.eye_pos;
+        g_regina_ctx.pass_constants.render_target_size = glm::vec2(window_width, window_height);
+        g_regina_ctx.pass_constants.inv_render_target_size = glm::vec2(1.0f / window_width, 1.0f / window_height);
+        g_regina_ctx.pass_constants.near_z = globals::default_depth_info().near_plane;
+        g_regina_ctx.pass_constants.far_z = globals::default_depth_info().far_plane;
+        g_regina_ctx.pass_constants.delta_time = globals::frame_info().delta_time().to_seconds();
+
+        renderer::commands::UpdateConstantBufferCommandDesc update_constant_buffer_command_desc;
+        update_constant_buffer_command_desc.element_index = 0;
+        update_constant_buffer_command_desc.buffer_data = memory::make_blob((rsl::byte*)&g_regina_ctx.pass_constants, rsl::memory_size(sizeof(PassConstants)));
+
+        renderer::update_constant_buffer(rsl::move(update_constant_buffer_command_desc), curr_pass_cb);
+    }
+
+    //-------------------------------------------------------------------------
+    bool initialize()
+    {
+        REX_LOG(LogRegina, "Initializing Regina");
```

Comment:
I think we have a problem here. By the time we reach this point, DX is initialized, but nothing can be drawn yet, which is fine, but the Window is shown already. I think we need to change our initialization order a bit. You need a window for the swapchain, but the window doesn't have to be shown yet.